src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -204,13 +200,34 @@ public static Text of(Object... objects) {
                 } else {
                     // Unsupported TextAction
                 }
+
+                // Special content - does not use the specified Text formatting
+            } else if (obj instanceof TextRepresentable) {
+                builder.append(((TextRepresentable) obj).toText());
+            } else if (obj instanceof CatalogType) {
```

Comment:
Whether the `CatalogType` is `Translatable` or just a "Name" does not really matter, sice this only for the visiual represenetation. However there is one thing i would like to add that is not present in vanilla and Sponge yet and has never been in Bukkit as well.
If you shiftclick on the name of the `CatalogType` it will copy the Id to your text/command input.

* The main point is shall i force `CatalogType`s to behave like this/the same across all implementations?
  * If yes, then we have to either make `CatalogType` extends `TextRepresentable` or keep the special treatment in `Text.of(Object...)`
  * If no, then we could just make the implementations in Common of `CatalogType` extends `TextRepresentable`, without forcing this by the API.
* The other important point is, shall I/we push the implementation work to the Implementation or is the fallback in the API enough. In both cases it is possible to customize the `CatalogType` presentation by extending `TextRepresentable` /overwriting `toText()`. 
  * If rely on fallback: Then we need the special case in `Texts.of(Object...)`, but don't need to change anything in the impl
  * if rely on the impl: Then we don't need the special case, but have to implement `TextRepresentable` for all CatalogTypes, including types with Mixins.
* Last but not least if we do this for `CatalogType` shouldn't we do it for other things as well and thus create heavy overhead in the `Texts.of(Object...)` method, because of all the exceptional behaviour/if checks.